pkg/apis/config/v1beta1/defaults.go

```diff
@@ -115,6 +115,13 @@ func SetDefaultTargetLoadPackingArgs(args *TargetLoadPackingArgs) {
 	if args.TargetUtilization == nil || *args.TargetUtilization <= 0 {
 		args.TargetUtilization = &DefaultTargetUtilizationPercent
 	}
+	metricProviderType := string(args.MetricProvider.Type)
```

Comment:
Silencing the error doesn't sound like a good idea, as the user may be specifying the type with a typo.

I'd suggest assigning the type to DefaultMetricProviderType when it's empty, and error it out in terms of illegal type, in the validation logic.

You can refer to how NodeResourcesAllocatableArgs.mode is defaulted and validated.

Reply:
And I also realized you're following the way how `LoadVariationRiskBalancingArgs` is handled. I's suggest updating that as well.

Reply:
Fixed